pycti/utils/opencti_stix2_splitter.py

```diff
@@ -96,16 +93,91 @@ def by_dep_size(elem):
             return elem["nb_deps"]
 
         self.elements.sort(key=by_dep_size)
-        for entity in self.elements:
+
+        # Algorithm to re-bundle elements
+        # Bundle created group the relationships by dependencies
+        # This grouping will improve relations sharding
+        if relations_grouping:
+            cache_ids = {}
+            id_alias = {}
+            ids_equivalence = {}
+            working_elements = {}
+            grouped_elements = {}
+            for elem in self.elements:
+                if cache_ids.get(elem["id"]):
+                    # Prevent sending duplicate elements for ingest
+                    continue
+                nb_deps = elem["nb_deps"]
+                if elem["type"] == "relationship":
+                    source_ref = (
+                        ids_equivalence.get(elem["source_ref"], elem["source_ref"])
```

Comment:
I don't really understand what is ids_equivalence since it will always return elem["source_ref"], why not use directly elem["source_ref"] ?

Reply:
ids_equivalence doesnt always return elem["source_ref"].
It will return the equivalence in the dictionnary or elem["source_ref"] if not found

Reply:
how can it be different ? get method returns a default value if not found, but it doesn't store anything.

Reply:
I removed the grouping part from the PR for now. It will be reintroduce in a dedicated one.